demo/controller/build.gradle

```diff
@@ -0,0 +1,127 @@
+}
+
+application {
+    mainClassName = 'io.deephaven.demo.control.DhDemoServer'
+}
+
+run {
+    environment("IS_CONTROLLER", System.getenv("IS_CONTROLLER") ?: "true")
+    environment("COOKIE_DOMAIN", "localhost")
+    environment("ALLOW_ADDRESS_LOOKUP", "false")
+}
+
+// we use withType(JavaExec) to catch intellij-created run configurations
+tasks.withType(JavaExec).configureEach {
+    JavaExec exe ->
+        // see https://vertx.io/docs/vertx-web/java/
+        // setting this here ensures local gradle runs won't cache things,
+        // but running in a production container (not a gradle JavaExec), we let it cache
+        exe.systemProperty("vertxweb.environment", "dev")
+        exe.systemProperty("dh-helm-root", new File(projectDir, "helm").absolutePath)
+        exe.systemProperty("dh-helm-mode", "generate")
+        exe.systemProperty("dh-helm-target", new File(buildDir, "helm").absolutePath)
+        exe.environment("IS_CONTROLLER", System.getenv("IS_CONTROLLER") ?: "true")
+        if (org.gradle.internal.os.OperatingSystem.current().macOsX) {
+            // these not-supported-by-mac properties default to true in application.properties
+            exe.systemProperty("quarkus.http.tcp-quick-ack", "false")
+            exe.systemProperty("quarkus.http.tcp-cork", "false")
+            exe.systemProperty("quarkus.http.tcp-fast-open", "false")
+        }
+}
+
+dependencies {
+    implementation project(':demo-gcloud')
+
+    String quarkusGroup = findProperty('quarkusPlatformGroupId')
+    String quarkusArtifact = findProperty('quarkusPlatformArtifactId')
+    String quarkusVersion = findProperty('quarkusPlatformVersion')
+    implementation enforcedPlatform("${quarkusGroup}:${quarkusArtifact}:${quarkusVersion}")
+    implementation enforcedPlatform("${quarkusGroup}:quarkus-google-cloud-services-bom:${quarkusVersion}")
+
+    // The tools we will use to build our controller
+    implementation 'io.quarkus:quarkus-vertx-http'
+   // implementation 'io.quarkus:quarkus-resteasy-reactive-jackson'
+    implementation 'io.smallrye.reactive:smallrye-mutiny-vertx-web-client'
+
+    implementation 'io.quarkus:quarkus-container-image-docker'
+
+    implementation 'com.squareup.okhttp3:okhttp'
+    implementation 'com.google.guava:guava'
+    implementation 'commons-io:commons-io'
+
+    runtime "io.netty:netty-transport-native-epoll:$quarkusVersion:linux-x86_64"
+    runtime "io.netty:netty-transport-native-kqueue:$quarkusVersion:osx-x86_64"
+
+    testImplementation 'io.quarkus:quarkus-junit5'
+    testImplementation 'io.rest-assured:rest-assured'
+    testImplementation "junit:junit:4.13.2'"
+}
+
+test {
+    useJUnit()
+    // uncomment below to leave test machine alive to play with
+    // systemProperty("noClean", "true") // WIP: must comment whole line to merge
+}
+
+String launchScript = 'launchScript'
+tasks.register(launchScript).configure{
+    Task t ->
+        final String startScript = """#!/bin/bash
```

Comment:
Yes, anywhere with big chunks of bash should move to src/main/scripts or the like.  It's obviously easier when prototyping to not do that, trivial to change since it's just a gradle file.